plugins/node/opentelemetry-instrumentation-aws-sdk/src/services/ServiceExtension.ts

```diff
@@ -29,6 +29,10 @@ import {
 export interface RequestMetadata {
   // isIncoming - if true, then the operation callback / promise should be bind with the operation's span
   isIncoming: boolean;
+  // isStream - if true, then the response is a stream so the span should not be ended by the middleware.
+  // the ServiceExtension must end the span itself, generally by wrapping the stream and ending after it is
+  // consumed.
+  isStream?: boolean;
```

Comment:
Do you know if the GenAI SIG discussed/documented wanting this behaviour of ending the span after the full stream is consumed?   I've seen opinions vary when discussing HTTP streaming. See the guidance for HTTP client span duration here: https://github.com/open-telemetry/semantic-conventions/blob/main/docs/http/http-spans.md#http-client-span-duration

Is there any equivalent in the Python GenAI-related instrumentations?

Reply:
> While the duration could keep streaming out of it, that would then need to override the end time of that span with the earlier timestamp, which I don't think the JS SDK supports.

Correct, it does not support that. The (documented) behaviour then would be that the Span duration would be just up until the first response from the server, effectively [TTFB](https://en.wikipedia.org/wiki/Time_to_first_byte).

Reply:
Ah do you mean that we should use TTFB here? That would mean we couldn't record usage information though.

BTW, I realized that this might be closer to RPC than HTTP which has some specification for streaming

https://opentelemetry.io/docs/specs/semconv/rpc/rpc-spans/#message-event

> an event for each message sent/received on client and server spans SHOULD be created

I think this also implies the overall span is for the whole stream. FWIW python keeps the span for the entire stream too.

Reply:
> Ah do you mean that we should use TTFB here?

No, I did not mean to imply what this "should" do. I don't have a strong opinion one way or the other. 

The HTTP span guidance from https://github.com/open-telemetry/semantic-conventions/blob/main/docs/http/http-spans.md#http-client-span-duration says:

>  Because of the potential for confusion around this, HTTP client library instrumentations SHOULD document their behavior around ending HTTP client spans.

I only meant to say that *if it is decided to handle this stream* by having the span end just on the initial response that this should be documented.

> FWIW python keeps the span for the entire stream too.

Sounds good to me to have the intention for the JS instrumentation to be the same.